_test/check-exercises.sh

```diff
@@ -27,14 +34,28 @@ for exercise in exercises/*/tests; do
         done
 
         # Run the test and get the status
-        cargo test
+        if [ -n "$DENYWARNINGS" ]; then
+            # No-run mode so we see no test output.
+            # Compiler errors will still be shown though.
+            # By the way, at the time of writing --quiet actually isn't quiet!
+            # It runs tests and prints output. Not what we want.
+            # Wonder if that behavior is intentional???
+            set +e
+            cargo test --no-run
+            status=$?
+            if [ $status -ne 0 ]; then
+                # Instead of immediate exit, we keep going.
+                # This allows us to see all warnings.
+                exitcode=1
```

Comment:
subshells can't affect their parent shell's vars, of course. have to `return` it.